apps/extension/src/background/approvals/messages.ts

```diff
@@ -0,0 +1,67 @@
+import { Message } from "router";
+import { ROUTE } from "./constants";
+
+enum MessageType {
+  RejectTransfer = "reject-transfer",
+  SubmitApprovedTransfer = "submit-approved-transfer",
+}
+
+export class RejectTransferMsg extends Message<void> {
+  public static type(): MessageType {
+    return MessageType.RejectTransfer;
+  }
+
+  constructor(public readonly msgId: string) {
+    super();
+  }
+
+  validate(): void {
+    if (!this.msgId || this.msgId === "") {
+      throw new Error("msgId was not provided!");
+    }
+    return;
+  }
+
+  route(): string {
+    return ROUTE;
+  }
+
+  type(): string {
+    return RejectTransferMsg.type();
+  }
+}
+
+export class SubmitApprovedTransferMsg extends Message<void> {
+  public static type(): MessageType {
+    return MessageType.SubmitApprovedTransfer;
+  }
+
+  constructor(
+    public readonly msgId: string,
+    public readonly address: string,
+    public readonly password: string
+  ) {
+    super();
+  }
+
+  validate(): void {
+    if (!this.msgId || this.msgId === "") {
```

Comment:
I think `|| this.msgId === ""` will always be `true` if `!this.msgId` is `true` :)
So we can either remove right operand or make it more explicit by checking for null, undefined and ""